website/docs/feature-flag-tutorials/use-cases/a-b-testing.md

```diff
@@ -1,112 +1,239 @@
+Here are two code examples of collecting impression data in an application to send to Google Analytics:
 
-A/B tests are performed safely and strategically with extra safeguards when you automate your flags based on user activity and other metrics of your choice.
+Example 1
+
+```js
+unleash.on(UnleashEvents.Impression, (e: ImpressionEvent) => {
+    // send to google analytics, something like
+    gtag("event", "screen_view", {
+        app_name: e.context.appName,
+        feature: e.featureName,
+        treatment: e.enabled ? e.variant : "Control", // in case we use feature disabled for control
+    });
+});
+```
+
+Example 2
+
+```js
+unleash.on(UnleashEvents.Impression, (e: ImpressionEvent) => {
+    if (e.enabled) {
+        // send to google analytics, something like
+        gtag("event", "screen_view", {
+            app_name: e.context.appName,
+            feature: e.featureName,
+            treatment: e.variant, // in case we use a variant for the control treatment
+        });
+    }
+});
+```
+
+In these example code snippets, `e` references the event object from the impression data output. Map these values to plug into the appropriate functions that make calls to your analytics tools and data warehouses.
+
+In some cases like in Example 1, you may want to use the "disabled feature" state as the "Control group".
+
+Alternatively, in Example 2, you can expose the feature to 100% of users and use two variants: "Control" and "Test". In either case, the variants are always used for the "Test" group. The difference is determined by how you use the "Control" group.
+
+An advantage of having your feature disabled for the Control group is that you can use metrics to see how many of the users are exposed to experiment(s) in comparison to the ones that are not. If you use only variants (for both the test and control group), you may see the feature metric as 100% exposed and would have to look deeper into the variant to know how many were exposed.
+
+Here is an example of a payload that is returned from Google Analytics that includes impression event data:
+
+```js
+{
+    "client_id": "unleash_client"
+    "user_id": "uuid1234"
+    "timestamp_micros": "1730407349525000"
+    "non_personalized_ads": true
+    "events": [
+        {
+            "name":"select_item"
+            "params": {
+                "items":[]
+                "event":"screen_view"
+                "app_name":"myAppName"
+                "feature":"myFeatureName"
```

Comment:
I still have a question about how you would account for a variant having multiple flags at the same time. Like if I have  a multivariate test that looks at a new version of the a. Nav, b. signup form and c. Marketing copy. Each of these flags could on/off, so there are 9 total variants. how is this accounted for?

Reply:
@kwasniew helped me with breaking down the way we'd do it in unleash for multivariate tests:

"Usually each flag can have multiple variants so the relationship is the opposite. If we are talking multivariate testing I think you’d rather have a flag for nav, a flag for sign up form and a flag for marketing copy. Each of those flags would have as many variants as required by each feature. And you’d always record what combination you got for a given user context and send it to the analytics platform and how it converted."